experimental/offline_attestation/README.md

```diff
@@ -0,0 +1,111 @@
+# Offline Attestation
+
+The current Oak Functions Runtime implementation uses an interactive handshake
+for setting up an attested end-to-end encrypted channel between the client and
+the runtime. For some scenarios it might be preferable to avoid the handshake.
+
+The offline attestation approach splits the attestation process into two
+distinct parts:
+
+- Key generation and export: A single asymmetric key-pair is generated. The
+  public key is bound to an attestation report that provides evidence about the
+  enclave that generated the private key. The public key and attestation report
+  are then made availble to clients.
+- Encrypted communications: Clients use the public key (once they are satisfied
+  by the evidence that only appropriate enclaves would have access to the
+  private key) to encrypt requests to the server. Clients also provide
+  additional information (e.g. the client's public key) in the request to allow
+  the server to encrypt the response.
```

Comment:
Can a client send a pre-shared secret instead. So that both sides can create a shared key from it?
And we don't need to have bidirectional asymmetric encryption?

Reply:
Yes that would work as well, and for an implementation that uses lower-level crypto primitives that would probably be preferable for performance reasons. This implementation uses hybrid encryption both ways because Tink prevents us from exporting a generated AES key to send it to the server as part of the encrypted request.